knative-operator/pkg/common/telemetry/telemetry.go

```diff
@@ -0,0 +1,186 @@
+package telemetry
+
+import (
+	"context"
+	"sync"
+
+	"github.com/openshift-knative/serverless-operator/knative-operator/pkg/common"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	kafkasourcev1beta1 "knative.dev/eventing-contrib/kafka/source/pkg/apis/sources/v1beta1"
+	eventingsourcesv1beta1 "knative.dev/eventing/pkg/apis/sources/v1beta1"
+	servingv1 "knative.dev/serving/pkg/apis/serving/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+)
+
+var log = common.Log.WithName("telemetry")
+
+type Telemetry struct {
+	name            string
+	stop            chan struct{}
+	metricsSnapshot sync.Map
+	tc              *controller.Controller
+}
+
+func NewTelemetry(name string, mgr manager.Manager, objects []runtime.Object) (*Telemetry, error) {
+	t := &Telemetry{}
+	t.name = name
+	t.stop = make(chan struct{})
+	tc, err := newTelemetryController(name, objects, mgr, t)
+	if err != nil {
+		return nil, err
+	}
+	t.tc = tc
+	return t, nil
```

Comment:
```suggestion
	return &Telemetry{
	  name: name,
	  stop: make(chan struct{}),
	  tc: tc
}, nil
```

Reply:
Here I need to pass a telemetry ref to the controller, it too late.